monkey/agent_plugins/credentials_collectors/chrome/src/windows_credentials_database_processor.py

```diff
@@ -1,16 +1,168 @@
-from typing import Collection
+import logging
+import shutil
+import sqlite3
+import string
+import tempfile
+from pathlib import Path
+from typing import Collection, List, Optional, Tuple
 
-from common.credentials import Credentials
+from Cryptodome.Cipher import AES
+
+from common.credentials import Credentials, EmailAddress, Password, Username
 from common.types import Event
+from common.utils.code_utils import secure_generate_random_string
+from infection_monkey.utils.threading import interruptible_iter
 
 from .utils import BrowserCredentialsDatabasePath
+from .windows_decryption import win32crypt_unprotect_data
 
+logger = logging.getLogger(__name__)
 
-class WindowsCredentialsDatabaseProcessor:
-    def __init__(self):
-        pass
 
+class WindowsCredentialsDatabaseProcessor:
     def __call__(
         self, interrupt: Event, database_paths: Collection[BrowserCredentialsDatabasePath]
     ) -> Collection[Credentials]:
-        return []
+        credentials = []
+
+        for item in interruptible_iter(database_paths, interrupt):
+            database_path = item.database_file_path
+            master_key = item.master_key
```

Comment:
NIT: `database_path` gets used in exactly one place. I'm not sure what value adding a local variable is.
NIT: Same for `master_key`.